docs/forms.rst

```diff
@@ -1,16 +1,21 @@
-Forms
-=====
+Customizing Forms
+=================
 
-The following forms can be overridden as needed in order to:
+All forms are based on ``django.forms.Form`` class, any custom fields you add
+will require custom logic to save the data. Some forms do not explicitly call
+the built-in save method, below lists all forms used and the final method
+called where you would want to save your custom data.
 
-- Add extra fields for extra required information
-- Override save to add extra functionality on save
+It is recommended that you review the default form classes provided that you
+wish to extend or replace in order to understand the fields and logic already
+supplied.
 
-Overriding Save
----------------
+To add additional fields or override functionality, create a custom class and
+inherit from the matching form listed below to supply your customizations.
 
-If you decide to add fields to a form, you will need to
-manually save the custom fields' data.
+To replace a form completely, create a custom class and inherit
+from ``forms.Form``. You will need to reimplement the original forms methods
+you have overridden
```

Comment:
First of, I appreciate the effort you spent in trying to up the documentation. But, I am wondering whether or not the above truly helps in explaining things:

- "All forms are based on django.forms.Form" -- that holds for any Django project with forms.

- "Custom fields require custom logic" -- true for any project with forms, does not convey much information.

- "... recommended to review" -- while true, from a documentation point of view, not very helpful.

Reply:
Mentioning Form was to differentiate between Form and ModelForm types.

I suppose the rest was to inform someone that if they add custom fields, said data will not just automatically be saved. I had seen too many posts (not directly related to allauth) where its expected that data on the form is just magically saved somehow. That's more so with ModelForms than Form types.